controllers/kustomization_decryptor.go

```diff
@@ -148,20 +148,27 @@ func (kd *KustomizeDecryptor) ImportKeys(ctx context.Context) error {
 
 		var ageIdentities []string
 		for name, file := range secret.Data {
-			switch filepath.Ext(name) {
-			case ".asc":
-				keyPath, err := securejoin.SecureJoin(tmpDir, name)
-				if err != nil {
-					return err
-				}
+			if name == intkeyservice.SOPSCredentialsFileAzureKeyvault {
+				keyPath := filepath.Join(kd.homeDir, name)
 				if err := os.WriteFile(keyPath, file, os.ModePerm); err != nil {
 					return fmt.Errorf("unable to write key to storage: %w", err)
 				}
-				if err := kd.gpgImport(keyPath); err != nil {
-					return err
+			} else {
+				switch filepath.Ext(name) {
+				case ".asc":
+					keyPath, err := securejoin.SecureJoin(tmpDir, name)
+					if err != nil {
+						return err
+					}
+					if err := os.WriteFile(keyPath, file, os.ModePerm); err != nil {
+						return fmt.Errorf("unable to write key to storage: %w", err)
+					}
+					if err := kd.gpgImport(keyPath); err != nil {
+						return err
+					}
+				case ".agekey":
+					ageIdentities = append(ageIdentities, string(file))
```

Comment:
Think this can be simplified to:

```go
if name == intkeyservice.SOPSCredentialsFileAzureKeyvault {
	keyPath := filepath.Join(kd.homeDir, name)
	if err := os.WriteFile(keyPath, file, os.ModePerm); err != nil {
		return fmt.Errorf("unable to write '%s' to storage: %w", name, err)
	}
	continue
}
switch filepath.Ext(name) {
	case ".asc":
...
````